### PR TITLE
Collation

### PR DESCRIPTION
**Pull request by pumazi**
Provides functionality to collate a published work. This is based on changes in Connexions/cnx-archive#423.

Please remove the commit prefixed with :skull: before merging.

### CODE REVIEW

File: cnxpublishing/tests/use_cases.py [hunk 1]
```diff
@@ -589,7 +589,7 @@ def check_BOOK_in_archive(test_case, cursor):
                        "shortId": document_short_id,
                        "title": "Document One"}]}]}]}
     cursor.execute("""\
-SELECT tree_to_json(uuid::text, concat_ws('.',major_version, minor_version))
+ SELECT tree_to_json(uuid::text, concat_ws('.',major_version, minor_version), FALSE)
```
Comment (karenc) [inline cnxpublishing/tests/use_cases.py]: Leading whitespace?

**Comment by karenc** [comment]
Looks great!  I'm going to add my commits from #124 to this branch.

**Comment by reedstrm** [comment]
@karen good idea, that gives the whole picture

**Comment by reedstrm** [comment]
Is this an idempotent process? What happens if I invoke collate-content on an already collated book?

**Comment by pumazi** [comment]
Idempotent, probably not. The main functionality that does the collation doesn't need to be, because it's only ever going to run once during publish.

This isn't necessarily the case for the temporary view. This should not stop this merge from going through, since the temporary view has been authorized for use via ACL mechanisms and we'll have responsible people using the tools that hit it.

If we do intend to allow re-collating in the short-term, then making sure it's idempotent is important. I say in the short-term here, because re-collating should not be allowed for published content. If changes are needed, they should be in the form of a new version.

**Pull request merged**